app/controllers/api/v1/providers_controller.rb

```diff
@@ -1,36 +1,28 @@
 module Api
   module V1
     class ProvidersController < ApplicationController
+      include NextLinkHeader
+
       # Potential edge case:
       # It is possible for older updated_at values to written to the database after this API has been queried for changes. This would mean that these changes are missed when the client makes a subsequent request using the next-link.
       # Possible causes of older updated_at values:
       # - delay between c# calculating datetime.UtcNow and value being written to postgres
       # - clock drift between servers
       def index
-        page_size = params[:per_page] || 100
+        per_page = params[:per_page] || 100
+        changed_since = params[:changed_since]
+
         ActiveRecord::Base.transaction do
           ActiveRecord::Base.connection.execute('LOCK provider, provider_enrichment, site IN SHARE UPDATE EXCLUSIVE MODE')
-          @providers = Provider.changed_since(params[:changed_since]).limit(page_size)
+          @providers = Provider.changed_since(changed_since).limit(per_page)
         end
-        last_provider = @providers.last
 
-        response.headers['Link'] = if last_provider
-                                     next_link((last_provider.updated_at + 1.second).utc.iso8601, last_provider.id, page_size)
-                                   else
-                                     next_link(params[:changed_since], "", page_size)
-                                   end
+        next_link_header("from_provider_id", @providers.last, changed_since, per_page)
```

Comment:
:100:

Reply:
@davidgisbey and I were also really pleased with how clean this turned out :)